cmd/state-installer/installer.go

```diff
@@ -100,6 +102,17 @@ func (i *Installer) Install() (rerr error) {
 		return errs.Wrap(err, "Failed to copy installation files to dir %s. Error received: %s", i.path, errs.JoinMessage(err))
 	}
 
+	// Install the state service as an app if necessary
+	app, err := i.installSvcApp()
+	if err != nil {
+		return errs.Wrap(err, "Installation of service app failed.")
+	}
+
+	err = app.EnableAutostart()
```

Comment:
App and autostart should be separate concepts that have no awareness of each (ie. neither should import the other). Instead this linking happens in the consuming code.

So here you might do something like:

```
autostart.New(app.Path())
```